src/test/java/jenkins/advancedqueue/jobinclusion/strategy/JobInclusionJobPropertyTest.java

```diff
@@ -3,53 +3,238 @@
 
     @Before
-    public void setup() {
-        jobProperty = new JobInclusionJobProperty(true, "TestJobGroup");
-        descriptor = new JobInclusionJobProperty.DescriptorImpl();
+    public void setUp() throws Exception {
+        property = new JobInclusionJobProperty(true, "testGroup");
+        project = j.createFreeStyleProject("testFolder_" + testName.getMethodName());
+        build = j.buildAndAssertSuccess(project);
+        listener = new StreamBuildListener(new PrintStream(System.out), StandardCharsets.UTF_8);
+        launcher = new hudson.Launcher.LocalLauncher(StreamTaskListener.fromStdout());
+    }
+
+    @After
+    public void deleteProject() throws Exception {
+        project.delete();
+    }
+
+    @Test
+    public void getDescriptor() {
+        JobInclusionJobProperty.DescriptorImpl descriptor = property.getDescriptor();
+        assertNotNull(descriptor);
+        assertEquals("XXX", descriptor.getDisplayName());
+        assertTrue(descriptor instanceof JobInclusionJobProperty.DescriptorImpl);
+    }
+
+    @Test
+    public void getJobActions() {
+        // Setup: Add a job group to the PriorityConfiguration
+        PriorityConfiguration.get().getJobGroups().clear(); // Clear existing job groups
+        PropertyBasedJobInclusionStrategy.addJobGroup("testJobGroup", "testGroup");
+
+        // Retrieve job actions
+        ListBoxModel jobActions = PropertyBasedJobInclusionStrategy.getJobGroups();
+        assertNotNull(jobActions);
+        assertFalse(jobActions.isEmpty());
+
+        // Check the size of the list
+        assertEquals("Expected number of job actions", 1, jobActions.size());
+
+        // Check the properties of the first element
+        ListBoxModel.Option firstOption = jobActions.get(0);
+        assertNotNull(firstOption);
+        assertEquals("Expected display name", "testJobGroup", firstOption.name);
+        assertEquals("Expected value", "testGroup", firstOption.value);
     }
 
     @Test
-    public void getJobGroupNameTest() {
-        assertEquals("TestJobGroup", jobProperty.getJobGroupName());
+    public void prebuild() {
+        // Assuming prebuild performs some pre-build actions
+        assertTrue(property.prebuild(build, listener));
+
+        // Additional assertions to verify the state after prebuild
+        assertNotNull(build);
+        assertNotNull(listener);
+        assertEquals(
+                "testFolder_" + testName.getMethodName(), build.getProject().getName());
+    }
+
+    @Test
+    public void perform() throws IOException, InterruptedException {
+        // Assuming perform executes some actions
+        assertTrue(property.perform(build, launcher, listener));
+
+        // Additional assertions to verify the state after perform
+        assertNotNull(build);
+        assertNotNull(launcher);
+        assertNotNull(listener);
+        assertEquals(
+                "testFolder_" + testName.getMethodName(), build.getProject().getName());
+    }
```

Comment:
There are methods that are inherited from the parent object.  I'd rather not have tests of those methods in this class.  Leave that test to the parent object.  That will also allow you to remove the declarations of `launcher` and `listener` that are only used in these two tests.
```suggestion
```